.travis.yml

```diff
@@ -3,8 +3,13 @@ go:
     - 1.7
     - 1.8
     - tip
-script: go test -v -randtests 1000 ./...
+script:
+    - PKGS=$(go list ./... | grep -v /vendor/)
+    - GO_FILES=$(find . -iname '*.go' | grep -v /vendor/)
+    - test -z $(gofmt -s -l $GO_FILES)
+    - go vet $PKGS
+    - go test -v -race -randtests 1000 $PKGS
 matrix:
     fast_finish: true
     allow_failures:
-        - go: tip
\ No newline at end of file
+        - go: tip
```

Comment:
I'm not a fan of the trailing new line. That is pedantic as anything though, so it doesn't really matter one way or the other.